src/test/java/com/flightstats/hub/cluster/DistributedLockRunnerTest.java

```diff
@@ -41,146 +47,187 @@ static void setupCurator() throws Exception {
-            tryWait(50, waitForOne);
-            distributedLockRunner.runWithLock(lockable, "/LockPath", 1, TimeUnit.MINUTES);
+            Consumer<LeadershipLock> lockable = leadershipLock -> trackLock("lockable3", latch, () -> {
+                sleep(75);
+            });
+            tryWait(50, waitForTwo);
+            distributedLockRunner.runWithLock(lockable, "/Lock", 1, TimeUnit.MINUTES);
         });
 
         try {
             latch.await(2, TimeUnit.MINUTES);
         } catch (InterruptedException e) {
             fail("something went wrong waiting for all the locks to finish!");
         }
-        assertEquals("lockable2", lockList.get().get(0));
-        assertTrue(lockList.get().containsAll(newArrayList("lockable1", "lockable2", "lockable3")));
+        log.debug(lockTimes.toString());
+
+        assertLocksHappened("lockable1", "lockable2", "lockable3");
+        assertTrue(locksOverlap("lockable1", "lockable2") || locksOverlap("lockable2", "lockable3"));
+        assertFalse(locksOverlap("lockable1", "lockable3"));
     }
 
     private void tryWait(long sleepMillis, CountDownLatch latch) {
-        try { latch.await(sleepMillis, TimeUnit.MILLISECONDS); }
-        catch (InterruptedException e) {
+        try {
+            latch.await(sleepMillis, TimeUnit.MILLISECONDS);
+        } catch (InterruptedException e) {
             Thread.currentThread().interrupt();
+            log.warn("thread interrupted");
             fail(e.getMessage());
         }
     }
 
-    private void doAThing(long sleepMillis, String name, CountDownLatch latch) {
+    private void trackLock(String name, CountDownLatch latch, Runnable doTheThing) {
+        DateTime startTime = DateTime.now();
+        log.debug("{} starting at {}", name, startTime);
+
+        doTheThing.run();
+
+        DateTime endTime = DateTime.now();
+        log.debug("{} ending at {}", name, endTime);
+        lockTimes.put(name, Range.closed(startTime, endTime));
+
+        latch.countDown();
+    }
+
+    private void sleep(long sleepMillis) {
         try {
             Thread.sleep(sleepMillis);
         } catch (Exception e) {
             fail(e.getMessage());
         }
+    }
 
-        log.debug("Updating list {} at {}",name, DateTime.now().toString());
-        lockList.getAndUpdate(list -> {
-            list.add(name);
-            return list;
-        });
+    private boolean locksOverlap(String lockName1, String lockName2) {
+        return lockTimes.get(lockName1).isConnected(lockTimes.get(lockName2));
+    }
 
-        latch.countDown();
+    private void assertLocksHappened(String... lockNames) {
+        List<String> expectedLocks = Stream.of(lockNames).sorted().collect(toList());
+        log.debug(lockTimes.toString());
+        List<String> actualLocks = lockTimes.entrySet().stream().map(Map.Entry::getKey).sorted().collect(toList());
+        assertEquals(expectedLocks, actualLocks);
+    }
```

Comment:
Would be nice to be able to use something like hamcrest's assertThat(lockTimes.entrySet(), containsInAnyOrder(lockNames) for this sort of thing.

Reply:
does `containsAnyOrder` equal `equalsInAnyOrder` or is it truly just contains? I always have to look that up. I'm happy to do that, though!